lib/actions/mcubootTargetActions.js

```diff
@@ -0,0 +1,216 @@
+    isMcuboot,
+});
+
+export const portKnownAction = port => ({
+    type: MCUBOOT_PORT_KNOWN,
+    port,
+});
+
+export const processUpdateAction = (message, percentage, duration) => ({
+    type: MCUBOOT_PROCESS_UPDATE,
+    message,
+    percentage,
+    duration,
+});
+
+export const writingReadyAction = fileName => ({
+    type: MCUBOOT_WRITING_READY,
+    fileName,
+});
+
+export const writingCloseAction = () => ({
+    type: MCUBOOT_WRITING_CLOSE,
+});
+
+export const writingStartAction = () => ({
+
+    type: MCUBOOT_WRITING_START,
+});
+
+export const writingEndAction = () => ({
+    type: MCUBOOT_WRITING_END,
+});
+
+export const writingSucceedAction = () => ({
+    type: MCUBOOT_WRITING_SUCCEED,
+});
+
+export const writingFailAction = errorMsg => ({
+    type: MCUBOOT_WRITING_FAIL,
+    errorMsg,
+});
+
+export const openDevice = selectedDevice => dispatch => {
+    dispatch(targetActions.targetTypeKnownAction(CommunicationType.MCU, false));
+    dispatch(mcubootKnownAction(true));
+    dispatch(portKnownAction(selectedDevice['serialport.1'].comName));
+    dispatch(targetActions.updateTargetWritable());
+    dispatch(targetActions.loadingEndAction());
+};
+
+export const toggleMcuboot = () => (dispatch, getState) => {
+    const { port } = getState().app.target;
+    const { isMcuboot } = getState().app.mcuboot;
+
+    if (isMcuboot) {
+        dispatch(mcubootKnownAction(false));
+        dispatch(portKnownAction(null));
+    } else {
+        dispatch(mcubootKnownAction(true));
+        dispatch(portKnownAction(port.comName));
+    }
+
+    dispatch(targetActions.updateTargetWritable());
+};
+
+export const prepareUpdate = () => (dispatch, getState) => {
+    const filesLoaded = getState().app.file.loaded;
+    const fileName = Object.keys(filesLoaded)[0];
+    dispatch(writingReadyAction(fileName));
+};
+
+export const canWrite = () => (dispatch, getState) => {
+    // Disable write button
+    dispatch(targetActions.targetWritableKnownAction(false));
+    dispatch(warningActions.targetWarningRemoveAction());
+
+    // Check if mcu firmware is detected.
+    // If not, then return.
+    const { mcubootFilePath } = getState().app.file;
+    if (!mcubootFilePath) {
+        return;
+    }
+
+    // Check if target is MCU target.
+    // If not, then return.
+    const { isMcuboot } = getState().app.mcuboot;
+    if (!isMcuboot) {
+        return;
+    }
+
+    // Enable write button if all above items have been checked
+    dispatch(warningActions.targetWarningRemoveAction());
+    dispatch(targetActions.targetWritableKnownAction(true));
+};
+
+export const performUpdate = () => (dispatch, getState) => {
+    const { port } = getState().app.mcuboot;
+    const { mcubootFilePath } = getState().app.file;
+    const noSerialNumber = -1;
+    const baudrate = 115200;
+    const timeout = 12000;
+    let totalPercentage = 0;
+    let totalDuration = 0;
+
+    const progressCallback = progress => {
+        let process;
```

Comment:
Please rename `process`, it's an essential global object.

Reply:
Fixed.